libvast/src/detail/process.cpp

```diff
@@ -89,8 +90,10 @@ static caf::settings get_settings_mach() {
     return {};
   struct task_basic_info t_info;
   mach_msg_type_number_t t_info_count = TASK_BASIC_INFO_COUNT;
-  task_info(task, TASK_BASIC_INFO, (task_info_t) &t_info, &t_info_count);
-  result["current-memory-usage"] = t_info.resident_size;
+  task_info(task, TASK_BASIC_INFO, (task_info_t)&t_info, &t_info_count);
+  // http://web.mit.edu/darwin/src/modules/xnu/osfmk/man/task_basic_info.html
+  // says the resident set size is counted in pages, so we multiply accordingly.
+  result["current-memory-usage"] = t_info.resident_size * PAGE_SIZE;
```

Comment:
I can confirm the reported value looks about correct.

Would it make sense to use `getpagesize()` to determine the value at runtime rather than using this macro? Not sure if it can be changed.

Reply:
According to https://github.com/giampaolo/psutil/issues/1906 `getpagesize()` is deprecated in Big Sur.